projectq/ops/_basics.py

```diff
@@ -203,20 +203,15 @@ def __or__(self, qubits):
         cmd = self.generate_command(qubits)
         apply_command(cmd)
 
-    @property
-    def matrix(self):
-        raise AttributeError("BasicGate has no matrix property. Use MatrixGate instead.")
-
-    @matrix.setter
-    def matrix(self, matrix=None):
-        raise AttributeError("You cannot set the matrix property of gates derived"
-                             "from BasicGate. Please use MatrixGate instead if"
-                             "you need that functionality.")
-
     def __eq__(self, other):
         """ Return True if equal (i.e., instance of same class).
```

Comment:
Maybe better "Return True if instance of the same class, unless other is an instance of :class:`MatrixGate`, in which case equality is to be checked by testing for existence and (approximate) equality of matrix representations."